### ISSUE TITLE
XTCReader and TRRReader do not correctly seek with trajectory[frame] (on macOS)

### ISSUE THREAD

**Issue by orbeckst** [MDAnalysis/mdanalysis#1942]
### Expected behaviour
```python
u.trajectory[frame]
```
should seek to frame when trajectory is a Gromacs XTC or TRR file.

### Actual behaviour
The `u.trajectory.frame` counter is set to frame but the actual trajectory frame is just forwarded (see below).

This a fairly nasty bug because MDAnalysis reports wrongly what it does. It will almost certainly lead to wrong results if people try to compare specific frames.

### Code to reproduce the behaviour

``` python
import MDAnalysis as mda
from MDAnalysis.tests.datafiles import TPR, XTC, TRR

u = mda.Universe(TPR, XTC)

u.trajectory.rewind()
for ii in range(10):
    u.trajectory[0]
    print(ii, u.trajectory.ts.frame, u.trajectory.ts.time)

```
outputs
```
0 0 100.00000762939453
1 0 200.00001525878906
2 0 300.0
3 0 400.0000305175781
4 0 500.0000305175781
5 0 600.0
6 0 700.0000610351562
7 0 800.0000610351562
8 0 900.0000610351562

---------------------------------------------------------------------------
StopIteration                             Traceback (most recent call last)
....
..../MDAnalysis/mdanalysis/package/MDAnalysis/lib/formats/libmdaxdr.pyx in MDAnalysis.lib.formats.libmdaxdr.XTCFile.read()

StopIteration: 
```

EDIT: Identical behavior is observed with either XTC or TRR.

Note that everything works as expected for the DCDReader
```python
u2 = mda.Universe(PSF, DCD)

u2.trajectory.rewind()
for ii in range(10):
    u2.trajectory[0]
    print(ii, u2.trajectory.ts.frame, u2.trajectory.ts.time)
```
which outputs
```
0 0 0.9999999119200186
1 0 0.9999999119200186
2 0 0.9999999119200186
3 0 0.9999999119200186
4 0 0.9999999119200186
5 0 0.9999999119200186
6 0 0.9999999119200186
7 0 0.9999999119200186
8 0 0.9999999119200186
9 0 0.9999999119200186
```

### Currently version of MDAnalysis:
(run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
0.18.1-dev

**Comment by tylerjereddy** [comment]
This would indeed be a disturbing issue, but like @kain88-de this works just fine for me locally with the latest checkout of develop.

**Comment by orbeckst** [comment]
- With 0.18.0 (on macOS, Python 3.6): same output as you https://github.com/MDAnalysis/mdanalysis/issues/1942#issuecomment-397732429
- With 0.18.1-dev at 518091f7a2359660897365677213e25d83499ba6 (on macOS, Python 3.6): my reported failure https://github.com/MDAnalysis/mdanalysis/issues/1942#issue-332867750

I am checking now Linux and will update this comment.

**Comment by kain88-de** [comment]
when I'm at the noted commit from you it works for me on my machine.

**Comment by kain88-de** [comment]
@orbeckst can you maybe write a test for this. We can then check if it fails on travis?

I think the osx test might have been failing for a while now on osx

**Comment by kain88-de** [comment]
https://travis-ci.org/MDAnalysis/mdanalysis/builds/386033817

Nevermind the osx builds are actually stalling. I have no idea when this started to happen.

**Comment by orbeckst** [comment]
I can write a test (hopefully later today...). This seems a prime opportunity for `git bisect` ;-)

**Comment by kain88-de** [comment]
if you use emacs and magit have a short file with the test. magit can run bisect with the file as a script and automatically find the offending commit for you. You just need to start the process.

**Comment by orbeckst** [comment]
```python
import sys
import numpy as np
import MDAnalysis as mda
from MDAnalysis.tests.datafiles import TPR, XTC, TRR

print("MDAnalysis version", mda.__version__)

u = mda.Universe(TPR, XTC)

results = []

u.trajectory.rewind()
try:
    for ii in range(10):
        u.trajectory[0]
        results.append([ii, u.trajectory.ts.frame, u.trajectory.ts.time])
except StopIteration:
    print("Wrongly stopped iterating")
    print("Output:\n>> {}".format("\n>> ".join([str(line) for line in results])))
    print("Test FAILED")
    sys.exit(1)

results = np.asarray(results)
if not np.allclose(results[:, 2], 0.0):
    print("Wrong frames")
    print("Output:\n>> {}".format("\n>> ".join([str(line) for line in results])))
    print("Test FAILED")
    sys.exit(1)

print("No sign of Issue #1942: test PASSED")
sys.exit(0)
```

**Comment by orbeckst** [comment]
Commit 8572d796a95e92bbdaa827325a1ae82e1664dcba touched [package/MDAnalysis/lib/formats/src/xdrfile.c](https://github.com/MDAnalysis/mdanalysis/commit/8572d796a95e92bbdaa827325a1ae82e1664dcba#diff-191abe55db5f99cb05237a38919e619c). Perhaps @tylerjereddy @kain88-de have an idea why this might lead to failures on macOS.

**Comment by orbeckst** [comment]
XTC/TRR is currently broken on macOS – when I run
```
pytest -n 4  --disable-warnings coordinates/test_xdr.py
```
on _Darwin yngvi.3500.dhcp.asu.edu 16.7.0 Darwin Kernel Version 16.7.0: Fri Apr 27 17:59:46 PDT 2018; root:xnu-3789.73.13~1/RELEASE_X86_64 x86_64_ I get **45 failed, 113 passed, 39 warnings, 7 error in 15.60 seconds**.
